### PR TITLE
feat: response time timeseries

### PR DESCRIPTION
**Pull request by taro-yu**
## Description
This PR create the API to plot timeseries graph of response time.
This API supports three cases: 'best', 'worst', and 'all' 

response_time_timeseries_best
response_time_timeseries_worst
response_time_timeseries_all

## Related links
Jira: https://tier4.atlassian.net/browse/RT2-823

## Notes for reviewers
Please use this zip to check actions.
[checkfiles.zip](https://github.com/tier4/CARET_analyze/files/12550475/checkfiles.zip)

## Pre-review checklist for the PR author

- [x] I've confirmed the [contribution guidelines](https://github.com/tier4/caret/blob/main/.github/CONTRIBUTING.md).
- [x] The PR is ready for review.

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [x] The PR has been properly tested.
- [x] The PR has been reviewed.

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] (Optional) The PR has been properly tested with CARET_report verification.
- [ ] There are no open discussions or they are tracked via tickets.
- [ ] The PR is ready for merge.

After all checkboxes are checked, anyone who has write access can merge the PR.

### CODE REVIEW

File: src/caret_analyze/plot/visualize_lib/bokeh/util/hover.py [hunk 1]
```diff
@@ -208,6 +218,9 @@ def to_list(self) -> list[str]:
                           'publish_node_name', 'subscribe_node_name']
         elif isinstance(self._target_object, (Publisher, Subscription)):
             hover_keys = ['legend_label', 'node_name', 'topic_name']
+        elif isinstance(self._target_object, Path):
+            # hover_keys = ['legend_label']
```
Comment (rokamu623) [inline src/caret_analyze/plot/visualize_lib/bokeh/util/hover.py:222]: Delete unnecessary comments.

File: src/caret_analyze/plot/timeseries/response_time_timeseries.py [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+
+    def __init__(
+        self,
+        target_objects: Sequence[Path],
+        case: str
+    ) -> None:
+        super().__init__(target_objects)
+        self._case = case
+
+    def to_dataframe(self, xaxis_type: str = 'system_time') -> pd.DataFrame:
```
Comment (rokamu623) [inline src/caret_analyze/plot/timeseries/response_time_timeseries.py:37]: Please add CARET_doc.

Ref: https://github.com/tier4/CARET_analyze/blob/a9a0c424d27b1dee2323584ecb46fbfa132547c0/src/caret_analyze/plot/timeseries/frequency_timeseries.py#L37-L57

File: src/caret_analyze/plot/timeseries/response_time_timeseries.py [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+                inplace=True
+            )
+            # TODO: Multi-column DataFrame are difficult for users to handle,
+            #       so it should be a single-column DataFrame.
+            response_df = self._add_top_level_column(response_df, to)
+            all_df = pd.concat([all_df, response_df], axis=1)
+
+        return all_df.sort_index(level=0, axis=1, sort_remaining=False)
+
+    def to_timeseries_records_list(
```
Comment (rokamu623) [inline src/caret_analyze/plot/timeseries/response_time_timeseries.py:57]: Please add CARET_doc.

Ref: https://github.com/tier4/CARET_analyze/blob/a9a0c424d27b1dee2323584ecb46fbfa132547c0/src/caret_analyze/plot/timeseries/frequency_timeseries.py#L80-L95

**Review by rokamu623 — CHANGES_REQUESTED** [review]
There are a number of minor points that need to be corrected.

File: src/caret_analyze/plot/visualize_lib/bokeh/timeseries.py [hunk 4]
```diff
@@ -55,15 +57,22 @@ def create_figure(self) -> Figure:
...
         elif isinstance(target_objects[0], Communication):
             title = f'Time-line of communications {y_axis_label}'
+        elif isinstance(target_objects[0], Path):
+            title = f'Time-line of Paths {y_axis_label} --- {self._case} case ---'
         else:
             title = f'Time-line of publishes/subscribes {y_axis_label}'
         fig = init_figure(title, self._ywheel_zoom, self._xaxis_type, y_axis_label)
+        # if self._case is not None:
+        #     hover = HoverTool(tooltips=[("X軸", "@x"), ("Y軸", "@y")])
+        #     fig.add_tools(hover)
```
[the remark this answers is not in the document]
Comment (taro-yu) [reply src/caret_analyze/plot/visualize_lib/bokeh/timeseries.py]: Fixed in https://github.com/tier4/CARET_analyze/pull/322/commits/f4cd2af6e1f243a8c68186f3341424cb97eae919

File: src/caret_analyze/plot/visualize_lib/bokeh/util/hover.py (hunk 1 above)
Comment (taro-yu) [reply src/caret_analyze/plot/visualize_lib/bokeh/util/hover.py]: Fixed in https://github.com/tier4/CARET_analyze/commit/f4cd2af6e1f243a8c68186f3341424cb97eae919

File: src/caret_analyze/plot/timeseries/response_time_timeseries.py (hunk 2 above)
Comment (taro-yu) [reply src/caret_analyze/plot/timeseries/response_time_timeseries.py:37]: Fixed in https://github.com/tier4/CARET_analyze/commit/f4cd2af6e1f243a8c68186f3341424cb97eae919

File: src/caret_analyze/plot/timeseries/response_time_timeseries.py (hunk 3 above)
Comment (taro-yu) [reply src/caret_analyze/plot/timeseries/response_time_timeseries.py:77]: Fixed in https://github.com/tier4/CARET_analyze/commit/f4cd2af6e1f243a8c68186f3341424cb97eae919

**Review by rokamu623 — APPROVED** [review]
LGTM!

Make the appropriate reproduction code and have it checked.

**Review by rokamu623 — DISMISSED** [review]
LGTM!

**Review by nabetetsu — APPROVED** [review]
I have confirmed that the PR works as exppected as below results.
LGTM :+1: 
[rt2-823-result.zip](https://github.com/tier4/CARET_analyze/files/12584425/rt2-823-result.zip)

**Pull request merged**